index.d.ts

```diff
@@ -0,0 +1,55 @@
+interface IWickedElementsComponent<A extends Array<string>> {
+  /**
+   * Always triggered once a node is live (even with classes).
+   * Right before `onconnected` and only once,
+   * ideal to setup anything as a one off operation.
+   */
+  init(e: Event): void;
+  /**
+   * Triggered once live.
+   * If defined later on and already live it will trigger once (setup here).
+   */
+  onconnected?(e: Event): void;
+  /**
+   * Triggered once lost/removed.
+   */
+  ondisconnected?(e: Event): void;
+  /**
+   * Triggered when an attribute in the `attributeFilter` property changes,
+   * or, if said property is not defined or is an empty array, any time an
+   * attribute changes. 
+   */
+  onattributechanged?(e: {
+    attributeName: A[number];
```

Comment:
I think this is a `string` ?

Reply:
Yes, we could just type `attributeName` as a string.
But doing this allows you to narrow it to the values contained in the `A` tuple.

![neat-tipyngs](https://user-images.githubusercontent.com/26189644/66559215-32f80600-eb7f-11e9-939b-4d8189635292.png)

Furthermore, if you don't specify any `attributeFilter` it will also work.

Reply:
I see, thanks